sno/sno_context.py

```diff
@@ -0,0 +1,43 @@
+import click
+import pygit2
+import os
+
+
+class SnoContext(object):
+    def __init__(self):
+        self._repo_dir = None
+
+    @property
+    def repo_dir(self):
+        """The path of the repository. Defaults to the current directory."""
+        return self._repo_dir or os.curdir
+
+    @repo_dir.setter
+    def repo_dir(self, repo_dir):
+        self._repo_dir = repo_dir
+        if hasattr(self, "_repo"):
+            del self._repo
+
+    @property
+    def repo_dir_is_set(self):
```

Comment:
In the style of is/has, maybe:
```suggestion
    def has_repo_dir(self):
```
?

Reply:
Not quite sure about this name since obj.has_repo_path is False even though obj.repo_path is never None. But, done.